core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -460,31 +460,22 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
-    boolean connectDetailsMatch = connectDetailsMatch(mUri, conf);
-
-    if (sInitialized && connectDetailsMatch) {
-      updateFileSystemAndContext();
-      return;
-    }
     synchronized (INIT_LOCK) {
-      // If someone has initialized the object since the last check, return
       if (sInitialized) {
-        if (connectDetailsMatch) {
-          updateFileSystemAndContext();
-          return;
-        } else {
+        if (!connectDetailsMatch(mUri, conf)) {
           LOG.warn(ExceptionMessage.DIFFERENT_MASTER_ADDRESS
               .getMessage(mUri.getHost() + ":" + mUri.getPort(),
                   FileSystemContext.get().getMasterAddress()));
-          sInitialized = false;
+          initializeInternal(uri, conf);
```

Comment:
further consolidate the two `initializeInternal`?

Reply:
I tried but I think the code ends up messier since then we can't inline `connectDetailsMatch`

Reply:
sG